driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnection.java

```diff
@@ -410,26 +411,36 @@ private void sendCommandMessage(final CommandMessage message,
     private <T> T receiveCommandMessageResponse(final Decoder<T> decoder,
                                                 final CommandEventSender commandEventSender, final SessionContext sessionContext,
                                                 final int additionalTimeout) {
-        try (ResponseBuffers responseBuffers = receiveMessageWithAdditionalTimeout(additionalTimeout)) {
+        ResponseBuffers responseBuffers;
+        try {
+            responseBuffers = receiveMessageWithAdditionalTimeout(additionalTimeout);
+        } catch (Exception e) {
+            commandEventSender.sendFailedEvent(e);
+            throw e;
+        }
+
+        try {
             updateSessionContext(sessionContext, responseBuffers);
             if (!isCommandOk(responseBuffers)) {
-                throw getCommandFailureException(responseBuffers.getResponseDocument(responseTo, new BsonDocumentCodec()),
-                        description.getServerAddress());
+                MongoException commandFailureException = getCommandFailureException(responseBuffers.getResponseDocument(responseTo,
+                                new BsonDocumentCodec()), description.getServerAddress());
+                commandEventSender.sendFailedEvent(commandFailureException);
```

Comment:
`sendFailedEvent` being present twice in the code does not smell good. I propose a simpler approach (it also requires less changes to the existing code): https://github.com/stIncMale/mongo-java-driver/commit/58f950c031ca3a772acaa70584618c2044570ff6.

Reply:
I agree that the complexity of the additional boolean local variable is worth the simplification.  Thanks for the suggestion.  

Done